build/unix/compiledata.sh

```diff
@@ -109,6 +117,7 @@ echo "#define BUILD_NODE \""`uname -a`"\"" >> ${COMPILEDATA}.tmp
 echo "#define CXX \"$BXX\"" >> ${COMPILEDATA}.tmp
 echo "#define COMPILER \""`type -path $CXX`"\"" >> ${COMPILEDATA}.tmp
 echo "#define COMPILERVERS \"$COMPILERVERS\"" >> ${COMPILEDATA}.tmp
+echo "#define COMPILERVERSSTR \"$COMPILERVERSSTR\"" >> ${COMPILEDATA}.tmp
```

Comment:
Can we not just use the compiler name and version provided by CMake in `${CMAKE_CXX_COMPILER_ID}` and `${CMAKE_CXX_COMPILER_VERSION}`? This also seems redundant with the two lines right above (`COMPILER` and `COMPILERVERS`).

Reply:
No that is not good enough as on macOS we have an Apple clang version and official clang via brew. This would not be expressed in the CMAKE compiler macros.